deployment/deploymentClass.js

```diff
@@ -0,0 +1,448 @@
+    if (this.type == "deploy" || this.type == "") {
+      scripts.push(this.getDeploymentScript(location));
+    } else {
+      scripts.push("graph build");
+    }
+
+    this.scripts.set(location, scripts);
+  }
+
+  // Just flattens the first level of the json file so you have networks as keys, instead of protocol type.
+  flattenFirstJsonLevel() {
+    let result = {};
+    Object.keys(this.data).forEach((key1) =>
+      Object.keys(this.data[key1]).forEach(
+        (key2) => (result[key2] = this.data[key1][key2])
+      )
+    );
+    this.data = result;
+  }
+
+  // Makes sure the input arguments for the deployments you want are sensible.
+  checkValidScope() {
+    if (!this.protocol && !this.fork) {
+      throw "--PROTOCOL, --NETWORK, --FORK: Please specify a protocol, protocol and network, or fork.";
+    } else if (this.fork && (this.protocol || this.network)) {
+      throw "--PROTOCOL, --NETWORK, --FORK: If you specify a fork, you must not specify a protocol or network";
+    }
+  }
+
+  // Checks if the input arguments are valid for the deployment location.
+  checkValidService() {
+    switch (this.service) {
+      case "subgraph-studio":
+      case "studio":
+      case "s":
+      case "decentralized-network":
+      case "d":
+        return true;
+      case "hosted-service":
+      case "hosted":
+      case "h":
+        return true;
+      case "cronos-portal":
+      case "cronos":
+      case "c":
+        return true;
+      default:
+        throw (
+          "--SERVICE: Service is not valid or is missing: service=" +
+          this.service
+        );
+    }
+  }
+
+  // Makes sure the version specified in the json file is valid.
+  checkValidVersion(version) {
+    this.checkVersionLengthIsTwo(version);
+    this.checkVersionIsNumber(version);
+  }
+
+  checkVersionLengthIsTwo(version) {
+    if (version.split(".").length - 1 != 2) {
+      throw "See deployment.json: (1) version is not valid - must be in format x.x.x (e.g. 1.3.1)";
+    }
+  }
+
+  checkVersionIsNumber(version) {
+    let array = version.split(".");
+    if (
+      !array.every((element) => {
+        return !isNaN(element);
+      })
+    ) {
+      throw "See deployment.json: (2) version is not valid - must be in format x.x.x (e.g. 1.3.1)";
+    }
+  }
+
+  // Checks if the protocol level data in the deployment json file is present and/or valid.
+  checkProtocolLevelData(protocol) {
+    if (this.data[protocol] == undefined) {
+      throw (
+        "Check --PROTOCOL spelling as argument or in deployment.json: protocol= " +
+        protocol
+      );
+    } else if (Object.keys(this.data[protocol]) == []) {
+      throw (
+        "See deployment.json: No networks are defined for: protocol=" + protocol
+      );
+    }
+  }
+
+  // Checks if the network level data necessary to build this subgraph is present and/or valid.
+  checkNetworkLevelBuildDataOnly(protocol, network) {
+    if (this.data[protocol]["networks"][network] == undefined) {
```

Comment:
I just considered that in all likelihood this would fail, because people forgot to add the key in the deployment.json. However, I see that a blank string is a reasonable check as well, so I will add it.